DXRando/DeusEx/Classes/DXRBarrelFire.uc

```diff
@@ -49,9 +49,78 @@ function Tick(float delta)
     Super.Tick(delta);
 }
 
+function Destroyed()
+{
+    local int i;
+	local Vector loc;
+	local TrashPaper trash;
+	local Fire fire;
+    local bool fireHasSpawned;
+
+	// trace down to see if we are sitting on the ground
+	loc = vect(0,0,0);
+	loc.Z -= CollisionHeight + 8.0;
+	loc += Location;
+
+	// only generate trash if we are on the ground
```

Comment:
again wondering if we should get rid of the ground check

Reply:
I think the check needs to remain for paper for the reason I described in trashcan_trashbags. The biggest issue is the way the paper crashes to the ground when it hits a wall, which is very notable when it's high up.